Source/Core/Core/HW/AudioInterface.cpp

```diff
@@ -338,10 +338,11 @@ static void Update(u64 userdata, s64 cycles_late)
 
 int GetAIPeriod()
 {
-  u64 period = s_cpu_cycles_per_sample * (s_interrupt_timing - s_sample_counter);
-  u64 s_period = s_cpu_cycles_per_sample * s_ais_sample_rate;
-  if (period == 0)
+  const u64 s_period = s_cpu_cycles_per_sample * s_ais_sample_rate;
+  if (s_interrupt_timing <= s_sample_counter)
```

Comment:
I'm not convinced we should fail silently and return a value that possibly doesn't make sense rather than add an assertion for `s_interrupt_timing >= s_sample_counter`.

Reply:
We don't fail here. If I understand the logic correctly, the point of this function is to return the shortest period before the next AI event.

If we've already reached the sample count value where it triggers an interrupt (i.e. when `AIINTVLD` is set) then the only event we handle is the next AI period. Otherwise, if we may reach that sample count because `s_sample_counter < s_interrupt_timing`, then we take the smaller period of those two.